machine_learning_hep/analysis/do_systematics.py

```diff
@@ -0,0 +1,1294 @@
+
+from machine_learning_hep.do_variations import (
+    format_varlabel,
+    format_varname,
+    healthy_structure,
+)
+from machine_learning_hep.logger import get_logger
+
+# HF specific imports
+from machine_learning_hep.utilities import (  # make_plot,
+    combine_graphs,
+    draw_latex,
+    get_colour,
+    get_marker,
+    get_plot_range,
+    get_y_window_gr,
+    get_y_window_his,
+    make_message_notfound,
+    reset_graph_outside_range,
+    reset_hist_outside_range,
+    setup_canvas,
+    setup_histogram,
+    setup_legend,
+    setup_tgraph,
+)
+from machine_learning_hep.utils.hist import get_axis, print_histogram
+
+
+def shrink_err_x(graph, width=0.1):
+    for i in range(graph.GetN()):
+        graph.SetPointEXlow(i, width)
+        graph.SetPointEXhigh(i, width)
+
+
+# final ranges
+x_range = {
+    "zg": [0.1, 0.5],
+    "rg": [0.0, 0.4],
+    "nsd": [-0.5, 5.5],
+    "zpar": [0.3, 1.0],
+}
+
+
+class AnalyzerJetSystematics:
+    def __init__(self, path_database_analysis: str, typean: str, var: str):
+        self.logger = get_logger()
+        self.typean = typean
+        self.var = var
+        self.method = "sidesub"
+        self.logger.setLevel(logging.INFO)
+
+        # self.logger.info("Processing observable %s", self.var)
+
+        with open(path_database_analysis, "r", encoding="utf-8") as file_in:
+            db_analysis = yaml.safe_load(file_in)
+        case = list(db_analysis.keys())[0]
+        datap = db_analysis[case]
+
+        # plotting
+        # LaTeX string
+        self.latex_hadron = datap["analysis"][self.typean]["latexnamehadron"]
+        self.latex_ptjet = "#it{p}_{T}^{jet ch}"
+        self.latex_obs = datap["analysis"][self.typean]["observables"][self.var]["label"]
+        self.latex_y = datap["analysis"][self.typean]["observables"][self.var]["label_y"]
+
+        # binning of hadron pt
+        self.edges_pthf_min = datap["analysis"][self.typean]["sel_an_binmin"]
+        self.edges_pthf_max = datap["analysis"][self.typean]["sel_an_binmax"]
+        # self.n_bins_pthf = len(self.edges_pthf_min)
+
+        # binning of jet pt
+        self.ptjet_name = "ptjet"  # name
+        # reconstruction level
+        self.edges_ptjet_rec = datap["analysis"][self.typean]["bins_ptjet"]
+        self.n_bins_ptjet_rec = len(self.edges_ptjet_rec) - 1
+        self.ptjet_rec_min = self.edges_ptjet_rec[0]
+        self.ptjet_rec_max = self.edges_ptjet_rec[-1]
+        self.edges_ptjet_rec_min = self.edges_ptjet_rec[:-1]
+        self.edges_ptjet_rec_max = self.edges_ptjet_rec[1:]
+        # generator level
+        self.edges_ptjet_gen = datap["analysis"][self.typean]["bins_ptjet"]
+        self.n_bins_ptjet_gen = len(self.edges_ptjet_gen) - 1
+        self.ptjet_gen_min = self.edges_ptjet_gen[0]
+        self.ptjet_gen_max = self.edges_ptjet_gen[-1]
+        self.edges_ptjet_gen_min = self.edges_ptjet_gen[:-1]
+        self.edges_ptjet_gen_max = self.edges_ptjet_gen[1:]
```

Comment:
Do we really need this extra indirection of going through member variables for config options instead of directly accessing them through `self.cfg(...)`?

Reply:
Do you mean e.g. `self.cfg("bins_ptjet")` instead of `datap["analysis"][self.typean]["bins_ptjet"]`